crates/outbound-networking/src/lib.rs

```diff
@@ -299,7 +299,20 @@ pub struct OutboundUrl {
 
 impl OutboundUrl {
     pub fn parse(url: impl Into<String>, scheme: &str) -> anyhow::Result<Self> {
-        let url = url.into();
+        let mut url = url.into();
+
+        // Ensure that the authority is url encoded. Since the authority is ignored after this,
+        // we can always url encode the authority even if it is already encoded.
+        if let Some(at) = url.find('@') {
+            let scheme_end = url.find("://").map(|e| e + 3).unwrap_or(0);
+            let userinfo = &url[scheme_end..at];
+
+            let encoded = urlencoding::encode(userinfo);
+            let prefix = &url[..scheme_end];
+            let suffix = &url[scheme_end + userinfo.len()..];
+            url = format!("{prefix}{encoded}{suffix}");
```

Comment:
This results in `OutboundUrl::original` being the modified string rather than the _original_ original.  As far as I can tell that's only used for the `Display` implementation so I'm not sure if this is a concern - it might be surprising if displayed in a message and the user went "but that's not what I wrote"?